server/src/controllers/BookingController.mjs

```diff
@@ -1,142 +1,82 @@
+    //req: {RoomID: , Customers: [{Name:, IdentityCard:, address:, Type:, }, {Name:, IdentityCard:, address:, Type:, }]}
+    try {
+      const { RoomId, Customers } = req.body;
+      const getCurrentDate = () => {
+        const now = new Date();
+        const year = now.getUTCFullYear();
+        const month = String(now.getUTCMonth() + 1).padStart(2, '0'); // Months are zero-indexed
+        const day = String(now.getUTCDate()).padStart(2, '0');
 
-		// return result
-		if (bookings.length === 0) {
-			return res
-				.status(StatusCodes.NOT_FOUND)
-				.send("booking not found");
-		}
-		return res.status(StatusCodes.OK).send(bookings);
-	},
+        return `${year}-${month}-${day}`;
+      };
+      // create new booking
+      const booking = await BookingModel.createBooking(
+        getCurrentDate(),
+        RoomId
+      );
+      //update room status
+      await RoomModel.updateRoom(RoomId, null, 0, null, null);
+      //get bookingId of the new booking
+      const newBookingId = await BookingModel.getTheNewestBookingId();
+      //get all customerIds
+      var CustomerIds = [];
 
-	// create a new booking
-	post: (req, res) => {
-		const { RoomID, BookingDate, CheckOutDate, Cost } = req.body;
-		if (!RoomID || !BookingDate || !CheckOutDate || !Cost) {
-			return res
-				.status(StatusCodes.BAD_REQUEST)
-				.send("Invalid data supplied");
-		}
+      //create new customer if not exist
+      for (const Customer of Customers) {
+        const { Name, IdentityCard, Address, Type } = Customer;
+        const CustomerId = await customerModel.getCustomerIdByIdentityCard(
+          IdentityCard
+        );
 
-		const BookingID = bookingData.length + 1;
-		bookingData.push({
-			BookingID: BookingID,
-			RoomID: RoomID,
-			BookingDate: BookingDate,
-			CheckOutDate: CheckOutDate,
-			Cost: Cost,
-		});
+        if (CustomerId.length === 0) {
+          await customerModel.CreateCustomer(Name, IdentityCard, Address, Type);
+          CustomerIds.push(
+            (await customerModel.getCustomerIdByIdentityCard(IdentityCard))[0]
+          );
+        } else {
+          CustomerIds.push(CustomerId[0]);
+        }
+      }
 
-		return res.status(StatusCodes.CREATED).send(bookingData);
-	},
+      //create new bookingCustomer for each customer
+      for (const CustomerId of CustomerIds) {
+        await BookingCustomerModel.CreateBookingCustomer(
+          newBookingId[0].BookingId,
+          CustomerId.CustomerId
+        );
+      }
+      return res.status(StatusCodes.CREATED).json(booking);
+    } catch (err) {
+      return res.status(StatusCodes.INTERNAL_SERVER_ERROR).send(err.message);
+    }
+  },
 
-	// update a booking
-	patch: (req, res) => {
-		const { body, bookingIndex } = req;
-		roomData[bookingIndex] = {
-			...bookingData[bookingIndex],
-			...body,
-		};
-		return res
-			.status(StatusCodes.OK)
-			.send(bookingData[bookingIndex]);
-	},
+  getAllCustomersInBooking: async (req, res) => {
```

Comment:
Currently, as pointed out, this function does not return any thing (with bookingId=101)_tested in the browser and in Postman_. 
**Please look into this**

Reply:
đã xử lý trường hợp nếu BookingId không tìm thấy
![image](https://github.com/user-attachments/assets/0e614908-6f9f-4d77-a365-c54a95730be9)